### ISSUE TITLE
`useless-break` flags necessary `break` in `case`

### ISSUE THREAD

**Issue by bauersimon** [mgechev/revive#1281]
**Describe the bug**
`useless-break` flags a `break` statement within a `case`, followed by a label

**To Reproduce**
On version `v1.7.0` (8ece20b0789c517bd3a6742db0daa4dd5928146d)

```shell
revive --config revive.toml demo/...
```

```toml
[rule.useless-break]
Enabled = true
```

```go
// demo/main/main.go
package main

import "fmt"

func main() {
	switch 1 {
	case 1:
		fmt.Println("foo bar")
		if false { // Add code that jumps to the label so it compiles.
			goto SOME_LABEL
		}
		break
	SOME_LABEL:
		panic(nil)
	}
}
```

**Expected behavior**
No lint error, since the `break` prevents the control flow from entering `SOME_LABEL`.

**Logs**
`demo/main/main.go:12:3: useless break in case clause`

**Desktop (please complete the following information):**
 - OS: `openSUSE Leap 15.6`
 - `go1.24.1 linux/amd64`

**Comment by chavacava** [comment]
Hi @bauersimon thanks for reporting the issue.
Handling the case is a matter of checking that the `break` statement is actually the last statement in the block before raising the failure.
(If you are comfortable with the `revive` code base you could propose a PR ;) )

**Comment by ccoVeille** [comment]
I have a question.

This issue made me think about the possible  issue that may exists in the unreachable_code linter.

I could have created an issue, but I prefer to ask first.

Do you think there could be an issue when a `goto LABEL` is used in a switch, and a `continue`, `return`, `exit`, or `fallthrough`is present ?

I have to computer with me to check it.

So here I'm asking for your point of view.

**Comment by chavacava** [comment]
@ccoVeille sure there are cases where un-structured control flow might make the rule erroneously point code as unreachable. The rule handles some cases like not raising a failure if the statement following the statement taking the control out is labelled.

It's possible to make these rules completely sound by computing the actual control flow of the methods under analysis but, in my opinion, the the gain in precision doesn't justify then complexity of such a computation. (I'm not sure `revive` should help people writing code with complicated control flow using `goto`s)

**Comment by ccoVeille** [comment]
> I'm not sure revive should help people writing code with complicated control flow using gotos

Me neither.

![goto.png](https://github.com/user-attachments/assets/6301beb0-d459-4308-aa97-a342a3f8dc05)
https://xkcd.com/292/

**Comment by ccoVeille** [comment]
@chavacava I'm unsure if you mean opening an issue for unreachable_code rule is OK or not. For me it's a backlog issue, if someone wants to work on it, they could open a PR

**Comment by bauersimon** [comment]
@chavacava thanks for fixing this. I am unfamiliar with the codebase so I did not feel comfortable tackling this.
